File: code/ex70.py
class Solution:
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """

        #Fibnacci array

        temp_1 = 1
        temp_2 = 1

        if n == 1:
            return 1

        for i in range(n - 1):
            temp = temp_1 + temp_2
            temp_2 = temp_1
            temp_1 = temp

        return temp

sol = Solution()
print(sol.climbStairs(2))

File: code/ex70.cpp
class Solution {
public:
    int climbStairs(int n) {
        int dp1 = 1, dp11 = 1;
        for (int i = 2; i <= n; i++) {
            int bdp1 = dp1, bdp11 = dp11;
            dp1 = bdp1 + bdp11;
            dp11 = bdp1;
        }
        return dp1;
    }
};